Assignment 3.Rmd

```diff
@@ -136,18 +144,84 @@ What patterns do you see in the plot?
 It would be useful to determine how many people are in each cluster. We can do this easily with dplyr.
 
 ```{r}
-K7 <- 
+K7 <- count(K4, cluster)
 ```
 
 Look at the number of people in each cluster, now repeat this process for 3 rather than 2 clusters. Which cluster grouping do you think is more informative? Write your answer below:
 
+I think 3 cluster is more informative because for although for both 2 and 3 clusters we can see the min and max range of the average motivation, 3 clusters gives us more information of inbetween the range.
+
+```{r}
+
+
+
+#Below is code for process of 3 cluster rather than 2 
+threeCluster <- read.csv("Class_Motivation.csv")
+threeCluster <- select(threeCluster, motivation1,motivation2,motivation3,motivation4,motivation5 )
+threeCluster <- na.omit(threeCluster)
+threeCluster <- scale(threeCluster)
+
+fit <- kmeans(threeCluster, 3)
+threeCluster = data.frame(threeCluster, fit$cluster)
+names(threeCluster) = c("1","2","3","4","5","cluster")
+threeCluster <- gather(threeCluster, "week", "motivation", 1:5)
+
+threeCluster <- threeCluster %>% 
+  group_by(week, cluster) %>% 
+  summarise(average = mean(motivation))
+
+threeCluster$week <- as.numeric(threeCluster$week)
+
+threeCluster$cluster <- as.factor(threeCluster$cluster)
+
+ggplot(threeCluster, aes(week, average, colour = cluster)) + geom_line() + xlab("Week") + ylab("Average Motivation")
+
+threeCluster <- count(threeCluster, cluster)
+
+```
+
+
 Once you have done this, save both of your plots to the Assignment 5 file. Create a Zotero item to index your new computer program (Assignment 5.rmd) in Zotero. Then commit your assignment, push it to your Github account and then Pull request your version to the original assignment version so I can see it.
 
+
+
 ##Part II
 
 Now, try to do the same for the data collected in class. Create two sets of clusters, the answers to the questions and regions where people grew up. 
 
+```{r}
+
+partTwo = read.csv("HUDK405018-cluster.csv")
+partTwo = partTwo[-1:-2,]
+partTwo = partTwo[-1:-2,]
+partTwo <- select(partTwo, 18:26)
+
+for(i in 3:9) {partTwo[,i] = as.numeric(partTwo[,i])}
+
+partTwo <- na.omit(partTwo)
+
+partTwoFit <- kmeans(partTwo, 2)
+
+partTwo = data.frame(partTwo, partTwoFit$cluster)
+
+new <- gather(partTwo, "question", "hour", 3:11)
+
+new$partTwoFit.cluster <- as.factor(new$partTwoFit.cluster)
+
+
+```
+```{r}
+
+
+```
 ##Part III
 
 Create a visualization that shows the overlap between the two groups of clusters you created in part III.
 
+```{r}
+
+ggplot(new, aes(x = LocationLatitude, y = LocationLongitude, colour = partTwoFit.cluster)) + geom_point() + xlab("Longitude") + ylab("Latitude")
+
```

Comment:
Nice plot, although I'm not sure it gives you a good idea of how much the two sets of clusters overlap?